deepinv/physics/generator/inpainting.py

```diff
@@ -30,15 +30,31 @@ def __init__(
         self.device = device
         self.rng = rng
 
-    def step(self, batch_size=1) -> dict:
+    def step(self, batch_size=1, input_mask=None) -> dict:
         r"""
         Create a bernoulli mask.
 
         :param int batch_size: batch_size.
+        :param torch.Tensor, None input_mask: optional mask to be split. If None, all pixels are considered. If not None, only pixels where mask==1 are considered.
         :return: dictionary with key **'mask'**: tensor of size (batch_size, *tensor_size) with values in {0, 1}.
         :rtype: dict
         """
-        mask = torch.zeros((batch_size, *self.tensor_size), device=self.device)
-        mask[torch.rand_like(mask) > self.split_ratio] = 0
+        if not isinstance(input_mask, torch.Tensor) or tuple(input_mask.shape) != (
```

Comment:
the second condition is not right I believe, what if there is a single mask (batch dim=1)?